### DISCUSSION TITLE
Query Regarding MONAI Label Support for MPS on Apple M-Series Chips in macOS

### DISCUSSION THREAD

**Discussion by wien2020** [Project-MONAI/MONAILabel#1598] [General]
Dear MONAI Label Team,

I am keen on exploring the potential of MONAI Label in a macOS environment, particularly on systems equipped with Apple's M-series chips. Given the unique capabilities of these chips and the MPS (Metal Performance Shaders) framework, I am curious about the compatibility and performance aspects.

Could you provide insights on whether MONAI Label supports the MPS devices on macOS with M-series chips? If so, are there any specific configurations or considerations to keep in mind to effectively leverage MPS for medical image processing and analysis tasks using MONAI Label?

Any guidance or documentation you could share on this matter would be greatly appreciated.

Thank you for your time and assistance.

Best regards,
wien2020

**Comment by diazandr3s** [comment]
Hi @wien2020,

Good question. 

MONAI and MONAI Label are being designed and optimized to run on NVIDIA GPUs.
However, it'll be interesting to see the performance of the Apple chips as well. Have you tried running a PyTorch model? See here: https://pytorch.org/blog/introducing-accelerated-pytorch-training-on-mac/

MONAI and MONAI Label are based in PyTorch. If you can run inference or train a PyTorch model on these chips, you should be able to run a MONAI Label model as well.

Please let us know,

**Comment by wien2020** [comment]
Dear diazandr3s,

Thank you very much for your insightful response. I deeply appreciate your guidance regarding the use of MONAI Label and its optimization for NVIDIA GPUs. Your suggestion to explore the performance of Apple chips with PyTorch models is particularly intriguing.

I have found MONAI Label to be a valuable tool in my work, and its continued development and versatility are something I look forward to. Following your advice, I plan to delve into the information provided on accelerated PyTorch training on Mac (https://pytorch.org/blog/introducing-accelerated-pytorch-training-on-mac/). This will hopefully give me a clearer understanding of how I can leverage MONAI Label more effectively on Apple's M series chips.

I'm excited about this exploration and will be sure to share any significant findings or experiences I encounter in this process.

Once again, thank you for your support and for the remarkable work being done with MONAI Label.

Best regards,
wien2020

**Comment by charlesvoxelwise** [comment]
Hi @wien2020
Wondering if you had made MONAI and MONAI Label work in Mac M2?

**Comment by lassoan** [comment]
It is mostly up to Apple to address this. NVIDIA recognized the importance of pytorch and having libraries like MONAI years ago and has been making huge investments into their development, and ensures that they work well with their products. Apple should do the same - everybody would benefit from it.

I don't know how this could be achieved, though. It is not very likely that some volunteers with the right skillset show up here and do the work, just because users ask for it. You could try to request this from Apple. I don't have high hopes, but anyway, if you posted a request about improving MONAI compatibility/performance with Apple products on some Apple forums then copy the link here, so that others who come across this discussion can vote on it, too.
